src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -24,36 +97,51 @@ std::optional<Shot> calcBestShotOnGoal(const Field &field, const Team &friendly_
                                        double radius)
 {
     std::vector<Robot> obstacles;
-    for (const Robot &enemy_robot : enemy_team.getAllRobots())
-    {
-        // Only add the robot to the obstacles if it is not ignored
-        if (std::count(robots_to_ignore.begin(), robots_to_ignore.end(), enemy_robot) ==
-            0)
-        {
-            obstacles.emplace_back(enemy_robot);
-        }
-    }
-    for (const Robot &friendly_robot : friendly_team.getAllRobots())
+    std::vector<Robot> all_robots;
+
+    size_t max_num_robots =
+        (enemy_team.numRobots() + friendly_team.numRobots()) <= 1
+            ? 1
+            : (enemy_team.numRobots() + friendly_team.numRobots() - 1);
```

Comment:
curious what this is doing here? does calcbest shot not work on an empty field? 

also nit: ternary operators are banned

Reply:
This is used for reserving vector sizes. Can't reserve 0.

ternary operators are banned D:

Reply:
gotcha, why do we reserve 1 less than the total num of max obstacles? `(enemy_team.numRobots() + friendly_team.numRobots() - 1)`?

Reply:
we don't count the shooting robot

Reply:
the pass generator cost function will evaluate calcBestShot on random points across the field, regardless of the presence of a friendly robot nearby, we definitely want to take the shooting robot into account

your pruning check earlier should filter out the shooting robot anyway right?

Reply:
oh ok ya ill remove the -1; didn't know those points existed

Reply:
done